build.rs

```diff
@@ -1,17 +1,27 @@
 // Copyright 2018 the Deno authors. All rights reserved. MIT license.
 
 // Run "cargo build -vv" if you want to see gn output.
-// TODO For the time being you must set an env var DENO_BUILD_PATH
-// which might be `pwd`/out/debug or `pwd`/out/release.
-// TODO Currently DENO_BUILD_PATH must be absolute.
 // TODO Combine DENO_BUILD_PATH and OUT_DIR.
 
 use std::env;
+use std::path::Path;
 use std::process::Command;
 
 fn main() {
+  // Cargo sets PROFILE to either "debug" or "release", which conveniently
+  // matches the build modes we support.
   let mode = env::var("PROFILE").unwrap();
-  let deno_build_path = env::var("DENO_BUILD_PATH").unwrap();
+
+  let repo_root_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
```

Comment:
Please add a TODO about removing usage of CARGO_MANIFEST_DIR. We should strive to reduce the amount of assumptions made about the build setup - in this case that we are able to write to this directory.

Reply:
The directory is not assumed to be writable. 

I'm using this as a rust equivalent of `__dirname`, so we can resolve DENO_BUILD_PATH *iff* it is a relative path. You could make DENO_BUILD_PATH absolute or set `DENO_BUILD_PATH=../out/dir` and it wouldn't try to write to the source dir at all.

Reply:
Maybe what's confusing you is that rusts `Path.join()` behaves more like nodejs' `path.resolve()` and not like `path.join()`.